nox/sessions.py

```diff
@@ -368,6 +372,18 @@ def conda_install(
         if not args:
             raise ValueError("At least one argument required to install().")
 
+        if self._runner.global_config.no_install:
+            if venv._reused:
+                logger.info(
+                    "Venv exists: skipping conda installation, as --no-install is set."
+                )
+                return None
+            else:
+                if not self._runner.global_config.reuse_existing_virtualenvs:
+                    logger.info(
+                        "Venv not created yet: ignoring --no-install and installing from conda."
+                    )
```

Comment:
You asked about the coverage error (`382->388`). I think that happens because you have no test for when `--reuse-existing-virtualenvs --no-install` is passed and a new virtualenv was created. That's what the branch `382->388` is about, anyway.

I would suggest to remove this warning for now:

```suggestion
```

Actually, what I had in mind was not a warning in every session about ignoring the option, but a warning once at startup about using `--no-install` without `--reuse-existing-virtualenvs`. But I see now that we don't have a logger yet when the options are parsed, and there is no prior art for warning about command line options either.

In the interest of moving this forward, I think omitting the warning is a good idea. The warning does not seem important, and less code is good :wink:

Reply:
Just a small clarification about Coverage output: `382->388` is *not* the region of lines 382 to 388. It is the jump from 382 to 388. That happens in the situation described above: `--no-install --reuse-existing-virtualenvs` plus no reuse i.e. new venv.

Reply:
Oh, that's super interesting, thanks for the extra detail! I'm not sure I would've figured out that the syntax was referring to a jump.